modules/ROOT/pages/subqueries/count.adoc

```diff
@@ -0,0 +1,295 @@
+[[count]]
+= COUNT subqueries
+
+A `COUNT` subquery expression can be used to count the number of rows returned by the subquery.
+
+Any non-writing query is allowed. `COUNT` subqueries differ from regular queries in that the final `RETURN` clause may be omitted,
+as any variable defined within the subquery will not be available outside of the expression,
+even if a final `RETURN` clause is used. One exception to this is that for a `DISTINCT UNION` clause, the `RETURN` clause is still mandatory.
+
+It is worth noting that the `MATCH` keyword can be omitted in subqueries in cases where the `COUNT` consists of only a pattern and an optional `WHERE` clause.
+
+[[count-syntax]]
+== Syntax
+
+[source, syntax]
+----
+COUNT LCURLY (regularQuery | matchMode? patternList whereClause?) RCURLY
```

Comment:
Is matchMode correct? why not just MATCH

Reply:
This was just taken from:  https://github.com/neo4j/cypher-language-support/blob/main/packages/language-support/src/antlr-grammar/CypherParser.g4